Src/ILGPU/Frontend/CodeGenerator/Arithmetic.cs

```diff
@@ -159,6 +159,26 @@ partial class CodeGenerator
             {
                 switch (kind)
                 {
+                    case BinaryArithmeticKind.Add:
+                    case BinaryArithmeticKind.Sub:
+                    case BinaryArithmeticKind.Mul:
+                        if (left.BasicValueType == BasicValueType.Int1 ||
+                            right.BasicValueType == BasicValueType.Int1)
+                        {
+                            var const0 = Builder.CreatePrimitiveValue(Location, 0);
```

Comment:
QUESTION: This forces both the `left` and `right` values to either 1 or 0. However, if only one of the values is `Int1`, the other value will also be forced to 1 or 0. This would appear to fail the assertion in [CreatePredicate](https://github.com/m4rs-mt/ILGPU/blob/bbe0747ff0b7fea8411e3fcf3713bf79f5dd383e/Src/ILGPU/IR/Construction/Predicate.cs#L34).

Is that the desired outcome?

Reply:
Indeed, this can lead to unintended side effects. To be compatible with the latest .Net updates, it is sufficient to force the conversion to `int32` values in case of both operands being `int1`. Let me change the `if` condition accordingly.